mailbox/backup/src/main/java/org/apache/james/mailbox/backup/zip/ZippedMailAccountIterator.java

```diff
@@ -0,0 +1,106 @@
+package org.apache.james.mailbox.backup.zip;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipException;
+
+import org.apache.james.mailbox.backup.MailArchiveEntry;
+import org.apache.james.mailbox.backup.MailArchiveIterator;
+import org.apache.james.mailbox.backup.MailboxWithAnnotationsArchiveEntry;
+import org.apache.james.mailbox.backup.SerializedMailboxId;
+import org.apache.james.mailbox.backup.UnknownArchiveEntry;
+import org.apache.james.mailbox.model.MailboxAnnotation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.fge.lambdas.Throwing;
+import com.google.common.collect.ImmutableList;
+
+public class ZippedMailAccountIterator implements MailArchiveIterator {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ZippedMailAccountIterator.class);
+    private static final List<MailboxAnnotation> NO_ANNOTATION = ImmutableList.of();
+    private final ZipEntryIterator zipEntryIterator;
+    private Optional<MailboxWithAnnotationsArchiveEntry> currentMailBox;
+    private Optional<ZipEntry> next;
+
+    public ZippedMailAccountIterator(ZipEntryIterator zipEntryIterator) {
+        this.zipEntryIterator = zipEntryIterator;
+        next = Optional.ofNullable(zipEntryIterator.next());
+    }
+
+    @Override
+    public void close() throws IOException {
+        zipEntryIterator.close();
+    }
+
+    @Override
+    public boolean hasNext() {
+        return next.isPresent();
+    }
+
+    @Override
+    public MailArchiveEntry next() {
+        return next.map(this::doNext).orElse(null);
+    }
+
+    private MailArchiveEntry doNext(ZipEntry currentElement) {
+        next = Optional.ofNullable(zipEntryIterator.next());
+        try {
+            return getMailArchiveEntry(currentElement);
+        } catch (Exception e) {
+            LOGGER.error("Error when reading archive on entry : " + currentElement.getName(), e);
+            next = Optional.empty();
+            return new UnknownArchiveEntry(currentElement.getName());
+        }
+    }
+
+    private MailArchiveEntry getMailArchiveEntry(ZipEntry currentElement) throws Exception {
+        Optional<ZipEntryType> entryType = ExtraFieldExtractor.getEntryType(currentElement);
+        return entryType
+            .map(Throwing.<ZipEntryType, MailArchiveEntry>function(type ->
+                from(currentElement, type)).sneakyThrow()
+            )
+            .orElseGet(() -> new UnknownArchiveEntry(currentElement.getName()));
+    }
+
+    private Optional<SerializedMailboxId> getMailBoxId(ZipEntry entry) throws ZipException {
+        return ExtraFieldExtractor.getStringExtraField(MailboxIdExtraField.ID_AM, entry).map(SerializedMailboxId::new);
+    }
+
+    private String getMailboxName(ZipEntry current) {
+        return current.getName().substring(0, current.getName().length() - 1);
```

Comment:
StringUtils.removeEnd?

Reply:
Nice i didn't know this, it's StringUtils.chop that i need !